apps/backend/src/reconciliation/reconciliation.service.ts

```diff
@@ -0,0 +1,84 @@
+import { POSDepositEntity } from './entities/pos-deposit.entity';
+import { TransactionEntity } from './entities/transaction.entity';
+import { Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { CashDepositEntity } from './entities/cash-deposits';
+
+@Injectable()
+export class ReconciliationService {
+  constructor(
+    @InjectRepository(TransactionEntity)
+    private salesRepo: Repository<TransactionEntity>,
+    @InjectRepository(POSDepositEntity)
+    private TDI34: Repository<POSDepositEntity>,
+    @InjectRepository(CashDepositEntity)
+    private TDI17: Repository<CashDepositEntity>
+  ) {}
+
+  //TODO ADD DTO AND RETURN TYPE
+  async addSales(body: any[]) {
+    return body.forEach(async (itm: any) => {
+      await this.salesRepo.save(this.salesRepo.create(itm));
+    });
+  }
+
+  //TODO ADD DTO
+  async addTDI34(body: any) {
```

Comment:
I think the promises are not handled properly here. 

Using async in a loop, do we need to await on Promise.all?

Reply:
if these are running sequentially then we need to make it parallel and use promise.all